vignettes/contribution.Rmd

```diff
@@ -0,0 +1,161 @@
+---
+title: "Package Structure"
+package: mmrm
+author:
+  - name: Liming Li
+output:
+  rmarkdown::html_document:
+          theme: "spacelab"
+          highlight: "kate"
+          toc: true
+          toc_float: true
+vignette: |
+  %\VignetteIndexEntry{Contributing Guide for Developers}
+  %\VignetteEncoding{UTF-8}
+  %\VignetteEngine{knitr::rmarkdown}
+editor_options:
+  chunk_output_type: console
+---
+
+# Introduction
+
+As `mmrm` package is built on `TMB` with complex structure, it might be hard for new developers to onboard.
+To make it easier for new developers to join, this documentation is created.
+
+# Package Structures
+
+`mmrm` is an R package, and the following sections describe all files/folders that are contained in this package.
+
+## data
+
+The `data/` folder is where `mmrm` internal data is stored.
+Currently, we only have `fev_data.rda` in this folder.
+In turn, `fev_data.rda` only contains the `fev_data` dataset.
+
+`fev_data` is a data frame of the following structure:
+
+```{r}
```

Comment:
probably need to call `data(fev_data, package = "mmrm")` or give a `library(mmrm)` at the top of this doc to deal with the error "error fev_data not found"